VAMobile/e2e/tests/utils.ts

```diff
@@ -129,38 +129,71 @@ export async function openDismissLeavingAppPopup(matchString: string, findbyText
  * @param newJsonValue - string or boolean: new value for the json object
  */
 
-export async function changeMockData (mockFileName: string, jsonProperty, newJsonValue: string | boolean) {
-			
-	fs.readFile('./src/store/api/demo/mocks/' + mockFileName, 'utf8', (error, data) => {
-		 if(error){
-			console.log(error);
-			return;
-		 }
-
-		const jsonParsed = JSON.parse(data)
-		var mockDataVariable
-		var mockDataKeyValue
-		for(var x=0; x<jsonProperty.length; x++) {
-			if (x == 0) {
-				mockDataVariable = jsonParsed[jsonProperty[x]]
-			} else if (x == jsonProperty.length - 1) {
-				mockDataVariable[jsonProperty[x]] = newJsonValue
-			} else {
-				if (jsonProperty[x].constructor == Object) {
-					var key = String(Object.keys(jsonProperty[x]))
-					var value = jsonProperty[x][key]
-					mockDataKeyValue = mockDataVariable[key]
-					mockDataVariable = mockDataKeyValue[value]
-				} else {
-					mockDataVariable = mockDataVariable[jsonProperty[x]]
-				}
-			}				
-		}
+export async function changeMockData (mockFileName: string, jsonProperty, newJsonValue, newEndpoint=false) {
+	
+  if(newEndpoint == true) {
```

Comment:
It looks like the only difference between the newEndpoint case and the default case is the mock directory. If so, instead of duplicating the function body, you can do something like:

```
const mockDir = isNewEndpoint ? './src/api/demoMocks/' : './src/store/api/demo/mocks/'
```

and then you can use that new variable when calling the readFile and writeFile functions.

I also recommend renaming newEndpoint to isNewEndpoint because boolean variables are easier to identify and read and they're prefixed with is/has/does, etc.

Reply:
Chris's suggestion makes sense. Although, in one of the recent React Query PRs, I think we agreed to keep all mock data together in the same directory. It looks like only 2 files are lingering in `VAMobile/src/api/demoMocks`. You could move those files into the `VAMobile/src/store/api/demo/mocks` directory, so you don't have to even make changes to this function. The only place the imports would need to be updated is [here](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/a492dec6de5f78b40d91ec75d6606f2d992265d9/VAMobile/src/store/api/demo/store.ts#L134-L135)